java/feathub-connectors/flink-connectors/flink-connector-prometheus/src/main/java/com/alibaba/feathub/flink/connectors/PrometheusSinkFunction.java

```diff
@@ -0,0 +1,342 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.feathub.flink.connectors;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;
+import org.apache.flink.table.catalog.Column;
+import org.apache.flink.table.catalog.ResolvedSchema;
+import org.apache.flink.table.data.ArrayData;
+import org.apache.flink.table.data.DecimalData;
+import org.apache.flink.table.data.MapData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.ArrayType;
+import org.apache.flink.table.types.logical.DecimalType;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.LogicalTypeFamily;
+import org.apache.flink.table.types.logical.LogicalTypeRoot;
+import org.apache.flink.table.types.logical.MapType;
+import org.apache.flink.util.StringUtils;
+
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.exporter.PushGateway;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.alibaba.feathub.flink.connectors.PrometheusConfigs.DELETE_ON_SHUTDOWN;
+import static com.alibaba.feathub.flink.connectors.PrometheusConfigs.EXTRA_LABELS;
+import static com.alibaba.feathub.flink.connectors.PrometheusConfigs.JOB_NAME;
+import static com.alibaba.feathub.flink.connectors.PrometheusConfigs.SERVER_URL;
+
+/**
+ * A {@link org.apache.flink.streaming.api.functions.sink.SinkFunction} that writes data to a
+ * Prometheus PushGateway. This SinkFunction supports numeric columns, list columns with numeric
+ * values, and map columns that map from strings to numeric values. Users can specify sets of labels
+ * for each column via the column comments. Label sets are separated by ';', labels within a set are
+ * separated by ',', and label names and values are separated by '='. For example:
+ * k1=v11,k2=v21;k1=v12,k2=v22.
+ *
+ * <p>The behavior of the Prometheus Sink for different column types is as follows:
+ *
+ * <ul>
+ *   <li>Numeric Column: Each column is emitted as a separate metric. Users can specify zero or one
+ *       label set for the metric to be emitted.
+ *   <li>List Column: Each element in the list is emitted as a separate metric. Users must specify
+ *       the same number of label sets as the number of elements in the list. Each label set
+ *       represents the corresponding element's labels. Additionally, all label sets must have the
+ *       same set of keys.
+ *   <li>Map Column: Each entry in the map is emitted as a separate metric. Users must specify
```

Comment:
How about using the following document:

The value type of this column must be a map<string, double>. For each key in this map, a metric is emitted with metric name equal to column's name, metric values equal to the values of the given key. The metric labels are derived from the column's comment as specified above, except that any label whose value is "null" in the column's comment should be updated to use the given key as the label value.

Note that it is not necessarily to explain the requirement of "having at least one label with null value" because 1) we will throw exception in this case with clear message and 2) for cases where user wants to use Map Column, they will always need to use null, and therefore they should not be surprised if we throw exception without documenting this. It is not necessary (and impossible) to document all possible exception in the Java doc.